### PR TITLE
libc: Implement memfd on top of tmpfs

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
https://man7.org/linux/man-pages/man2/memfd_create.2.html

## Impact
New API

## Testing
Pass CI

### CODE REVIEW

File: libs/libc/misc/lib_memfd.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int memfd_create(FAR const char *name, unsigned int flags)
+{
+#ifdef CONFIG_FS_TMPFS
+  char path[PATH_MAX];
+
+  snprintf(path, PATH_MAX, MEM_FD_VFS_PATH, name);
```
Comment (pkarashchenko) [inline libs/libc/misc/lib_memfd.c:45]:
```suggestion
  snprintf(path, size of(path), MEM_FD_VFS_PATH, name);
```

File: include/sys/memfd.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+#include <fcntl.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define MFD_CLOEXEC       O_CLOEXEC
+#define MFD_ALLOW_SEALING (1 << 16)
+#define MFD_HUGETLB       (1 << 17)
```
Comment (pkarashchenko) [inline include/sys/memfd.h:36]: I do not see how it is used. Maybe we can map it to 0?

Comment (xiaoxiang781216) [reply include/sys/memfd.h:36]: It could be implemented in the future if someone really need them.

File: libs/libc/misc/lib_memfd.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/misc/lib_memfd.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**